modules/apps/content-dashboard/content-dashboard-test/src/testIntegration/java/com/liferay/content/dashboard/web/internal/portlet/test/ContentDashboardAdminPortletGetPropsTest.java

```diff
@@ -0,0 +1,1138 @@
+					}));
+			JournalTestUtil.addArticle(
+				_group.getGroupId(), 0,
+				_getServiceContext(
+					TestPropsValues.getUserId(), _group.getGroupId(),
+					new long[] {childAssetCategory.getCategoryId()}));
+
+			Map<String, Object> data = _getData(
+				_getMockLiferayPortletRenderRequest(
+					new String[] {
+						assetVocabulary.getName(),
+						childAssetVocabulary.getName()
+					}));
+
+			Map<String, Object> props = (Map<String, Object>)data.get("props");
+
+			Assert.assertNotNull(props);
+
+			JSONArray vocabulariesJSONArray = (JSONArray)props.get(
+				"vocabularies");
+
+			Assert.assertEquals(
+				JSONUtil.putAll(
+					JSONUtil.put(
+						"categories",
+						JSONUtil.put(
+							JSONUtil.put(
+								"key",
+								String.valueOf(
+									childAssetCategory.getCategoryId())
+							).put(
+								"name", "child-category"
+							).put(
+								"value", 1L
+							).put(
+								"vocabularyName", "child-vocabulary"
+							))
+					).put(
+						"key", String.valueOf(assetCategory.getCategoryId())
+					).put(
+						"name", "category"
+					).put(
+						"value", 1L
+					).put(
+						"vocabularyName", "vocabulary"
+					)
+				).put(
+					JSONUtil.put(
+						"categories",
+						JSONUtil.put(
+							JSONUtil.put(
+								"key",
+								String.valueOf(
+									childAssetCategory.getCategoryId())
+							).put(
+								"name", "child-category"
+							).put(
+								"value", 1L
+							).put(
+								"vocabularyName", "child-vocabulary"
+							))
+					).put(
+						"key", "none"
+					).put(
+						"name", "No vocabulary Specified"
+					).put(
+						"value", 1L
+					).put(
+						"vocabularyName", "vocabulary"
+					)
+				).toString(),
+				vocabulariesJSONArray.toString());
+		}
+		finally {
+			_assetVocabularyLocalService.deleteAssetVocabulary(assetVocabulary);
+			_assetVocabularyLocalService.deleteAssetVocabulary(
+				childAssetVocabulary);
+		}
+	}
+
+	private Map<String, Object> _getData(
+			MockLiferayPortletRenderRequest mockLiferayPortletRenderRequest)
+		throws Exception {
+
+		MVCPortlet mvcPortlet = (MVCPortlet)_portlet;
+
+		mvcPortlet.render(
+			mockLiferayPortletRenderRequest,
+			new MockLiferayPortletRenderResponse());
+
+		return ReflectionTestUtil.invoke(
+			mockLiferayPortletRenderRequest.getAttribute(
+				"CONTENT_DASHBOARD_ADMIN_DISPLAY_CONTEXT"),
+			"getData", new Class<?>[0]);
+	}
+
+	private MockLiferayPortletRenderRequest _getMockLiferayPortletRenderRequest(
+			String[] assetVocabularyNames)
+		throws Exception {
+
+		return MockLiferayPortletRenderRequestUtil.
+			getMockLiferayPortletRenderRequest(
+				assetVocabularyNames, _company, LocaleUtil.getSiteDefault());
+	}
+
+	private ServiceContext _getServiceContext(
+			long userId, long groupId, long[] assetCategoryIds)
+		throws PortalException {
+
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext(groupId, userId);
+
+		serviceContext.setAssetCategoryIds(assetCategoryIds);
```

Comment:
Maybe we could create a new method ServiceContextTestUtil.getServiceContext with the assetCategoryIds. What do you think

Reply:
see https://github.com/liferay-tango/liferay-portal/pull/889#discussion_r576302341